.circleci/config.yml

```diff
@@ -38,6 +38,33 @@ aliases:
       keys:
         - v1-build-{{ .Revision }}
         - v1-build-
+  - &build_pr
+    <<: *defaults
```

Comment:
It looks like this grabs the `auth` section, mentioning `DOCKERHUB_*` variables, from the `defaults` section above. I'm not familiar with why we might need those... are they actually secret? Do they come from a context, and if so, will `build-pr` work correctly without these `DOCKERHUB_*` variables? If so, are they actually needed by other builds?

Reply:
We'll have to do something about this if ever Circle changes its docker hub policies, but is probably fine for now.  
They announced on November 21, 2020 that they won't rate limit access to docker hub through circle, but that agreement is subject to change.
Then we should probably use the context scratch-www-all.